src/libs/ReportUtils.ts

```diff
@@ -3537,6 +3544,17 @@ function getReportName(
     personalDetails?: Partial<PersonalDetailsList>,
     invoiceReceiverPolicy?: OnyxEntry<Policy>,
 ): string {
+    const reportID = report?.reportID;
+    const cacheKey = getCacheKey(report);
+
+    if (reportID) {
+        const reportNameFromCache = reportNameCache.get(cacheKey);
+
+        if (reportNameFromCache && reportNameFromCache.reportName === report?.reportName) {
```

Comment:
👋 Coming from https://github.com/Expensify/App/issues/47405
This has caused a small bug where DM header wouldn't show after you clear the cache, since `reportNameCache` is an empty string after you do it
We resolved this by returning  early if `reportNameFromCache.reportName` is not an empty string